pulsar/client/staging/up.py

```diff
@@ -482,7 +496,10 @@ def get_path():
                 if not name:
                     # TODO: consider fetching this from source so an actual input path
                     # isn't needed. At least it isn't used though.
-                    name = basename(path)
+                    if type == 'tool':
```

Comment:
No.

Reply:
No to hacking the path (line below) or checking `type == 'tool'`? Because you conditional `type == 'tool'` down below in the same method.